tests/sanity/ignore-2.18.txt

```diff
@@ -2,6 +2,7 @@ plugins/modules/consul_session.py validate-modules:parameter-state-invalid-choic
 plugins/modules/homectl.py import-3.11  # Uses deprecated stdlib library 'crypt'
 plugins/modules/homectl.py import-3.12  # Uses deprecated stdlib library 'crypt'
 plugins/modules/iptables_state.py validate-modules:undocumented-parameter             # params _back and _timeout used by action plugin
+plugins/modules/keycloak_userprofile.py validate-modules:parameter-invalid
```

Comment:
Why is this needed? Adding new entries here should be done with care, and this one looks like it is hiding a bug in the module.

Reply:
The concrete problem is the suboption `kc.user.profile.config` which is "so not python". I could of course abstract/rename this to be more pythonesque, but this is how it is named in keycloak's json representation.
Do you think it is valid to leave it or should it be changed?

Reply:
How about naming that option `kc_user_profile_config`, with an alias `kc.user.profile.config`? That way users can use both variants. The code will need to handle `kc_user_profile_config` though (and move its value to `kc.user.profile.config` when using `module.params['config']`), something like:
```
    config = module.params['config']
    if config:
        config = dict(config)  #  create a copy
        config['kc.user.profile.config'] = config.pop(`kc_user_profile_config`)
```
You're using `module.params['config']` (or its eqiuvalent `module.params.get('config')`) further down the code as well though, that needs to be adjusted to use this copy.

Reply:
Implemented your suggestion to name it `kc_user_profile_config` and now have aliases for a lot of (sub-)options.
Also added code that handles recursive camelCasel lists and dicts.

Reply:
I think it would be better not to have the name with the dots exposed to the Ansible user. It is not just the non- pythonesque aspect of it, but it would also create problems if that comes up in Jinja templating. Not to mention the need to create these exceptions in the sanity checking.
The problems outweigh the benefits, IMHO.

Reply:
Alright, removed the "doted" parameter option completely.